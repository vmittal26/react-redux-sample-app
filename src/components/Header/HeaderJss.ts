import styled from 'styled-components';

const HeaderWrapper = styled.div`
     background-color: #24292e;
     display: flex;
     top:0;
     position:fixed;
     width:100%;
     height: 2.5rem;
     align-items:center;
     z-index:1;
     padding:0 2rem;

     .home-button{
          outline:none;
          border:none;
          background-color:transparent;
          color:white;
          &:hover{
               color:rgba(255, 255, 255, 0.8);
          }
     }
     img{
          width:1.5rem;
          margin-left:auto;
          cursor: pointer;
          height:1.5rem;
     }
`;

export default HeaderWrapper;
