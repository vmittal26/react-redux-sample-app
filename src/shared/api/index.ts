export * from './getLoginAPIURL';
export * from './getSearchAPIURL';
