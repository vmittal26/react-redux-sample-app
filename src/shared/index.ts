export * from './api';
export * from './constants';
export * from './hooks/';
export * from './model';
export * from './styled-components';
export * from './utils';
