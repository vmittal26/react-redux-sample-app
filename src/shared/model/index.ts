export * from './AppState';
export * from './LoginTypes';
export * from './PlanetInfo';
export * from './SearchTypes';
export * from './ActionType';
