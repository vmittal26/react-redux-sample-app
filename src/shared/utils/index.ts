export * from './TestUtils';
export * from './sortPlanets';
