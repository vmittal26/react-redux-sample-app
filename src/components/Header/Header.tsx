import React, { useContext } from 'react';
import { useHistory } from 'react-router-dom';

import logo from '../../images/logout_blue.svg';
import { GlobalContext } from '../../shared';
import HeaderWrapper from './HeaderJss';


/**
 * Header Component
 */
export const Header: React.FC = () => {
  const [isLoggedIn, setIsLoggedIn] = React.useState(false);
  const { userContext } = useContext(GlobalContext);
  const history = useHistory();

  const onLogout = () => {
    userContext.setIsAuthenticated(false);
    history.push('/');
  };

  React.useEffect(() => {
    const unlisten = history.listen(() => {
      // on coming back to login page authentication flag should be set to false
      setIsLoggedIn(userContext.getIsAuthenticated());
    });

    return unlisten;
  }, [history, userContext]);

  return (
    <HeaderWrapper>
      <button className='home-button' onClick={onLogout}>Swapi React</button>
      {isLoggedIn && <img onClick={onLogout} src={logo} alt='swapi react'></img>}
    </HeaderWrapper>
  );
};
