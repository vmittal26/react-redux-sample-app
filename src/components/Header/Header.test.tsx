import { shallow } from 'enzyme';
import React from 'react';
import { useHistory } from 'react-router-dom';
import { getElementByAttr } from '../../shared';

import { Header } from './Header';


jest.mock('react-router-dom', () => ({
  useHistory: jest.fn(() => ({
    push: jest.fn(),
    goBack: jest.fn()
  }))
}));

const getProps = (isAuthenticated = false) => ({
  isAuthenticated,
  onLogout: jest.fn()
});
describe('Header', () => {
  const headerProps = getProps(true);
  const wrapper = shallow(<Header {...headerProps} />);
  const element = getElementByAttr(wrapper, 'logout-icon');

  beforeEach(() => {
    wrapper.setProps(headerProps);
  });
  test('renders learn Header correctly if user is authenticated ', () => {
    expect(element.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders learn Header correctly if user is not authenticated ', () => {
    wrapper.setProps(getProps());
    expect(wrapper).toMatchSnapshot();
  });

  test('button should have correct text', () => {
    const headerButton = getElementByAttr(wrapper, 'home-button');
    expect(headerButton.text()).toBe('Swapi React');
  });

  test('onlogout should be called on click of logout icon', () => {
    element.simulate('click');
    expect(headerProps.onLogout).toHaveBeenCalled();
    // expect(useHistory().push).toHaveBeenCalled();
  });

  // test('On press home should go to login page', () => {
  //   const homeButton = getElementByAttr(wrapper, 'home-button');
  //   homeButton.simulate('click');
  //   expect(useHistory().goBack).toHaveBeenCalled();
  // });
});
