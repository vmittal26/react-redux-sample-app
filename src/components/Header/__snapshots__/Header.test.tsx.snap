// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders learn Header correctly if user is authenticated  1`] = `
<styled.div>
  <button
    className="home-button"
    onClick={[Function]}
    test-id="home-button"
  >
    Swapi React
  </button>
  <img
    alt="swapi react"
    onClick={[Function]}
    src="logout_blue.svg"
    test-id="logout-icon"
  />
</styled.div>
`;

exports[`Header renders learn Header correctly if user is not authenticated  1`] = `
<styled.div>
  <button
    className="home-button"
    onClick={[Function]}
    test-id="home-button"
  >
    Swapi React
  </button>
</styled.div>
`;
